src/main/java/com/es/phoneshop/model/Product.java

```diff
@@ -17,6 +19,7 @@
     private Currency currency;
     private int stock;
     private String imageUrl;
+    private Map<LocalDate, BigDecimal> priceHistory;
```

Comment:
Probably it would be better to either:
- Change the type of this field to TreeMap (or even better to LinkedHashMap)
-- Pros: the entries are sorted by LocalDate comparator (or insert order) independently of logic which created the product
-- Cons: restricting the field type to the specific collection class (which is never cool)
- Let it be just a Map, but implement the sorting logic in ProductPriceHistoryServlet (instead of relying on the creation logic which should use TreeMaps)
-- Pros: we have just a happy Map field
-- Cons: it's assumed unsorted while being stored (i.e. it's not really a 'history')